hazelcast/src/hazelcast/client/xml_impl.cpp

```diff
@@ -0,0 +1,1317 @@
+            config.set_prefetch_count(
+              get_integer_value(pair.first, pair.second.data()));
+        } else if (matches("prefetch-validity-millis", node_name)) {
+            config.set_prefetch_validity_duration(std::chrono::milliseconds(
+              get_long_value(node_name, pair.second.data())));
+        }
+    }
+    client_config->add_flake_id_generator_config(config);
+}
+
+void
+client_dom_config_processor::handle_reliable_topic(
+  const boost::property_tree::ptree& node)
+{
+    handle_reliable_topic_node(node);
+}
+void
+client_dom_config_processor::handle_reliable_topic_node(
+  const boost::property_tree::ptree& node) const
+{
+    std::string name = get_attribute(node, "name");
+    hazelcast::client::config::reliable_topic_config config(name);
+    for (auto& pair : node) {
+        auto child = pair_to_node(pair.first, pair.second);
+        std::string node_name = pair.first;
+        if (matches("read-batch-size", node_name)) {
+            config.set_read_batch_size(
+              get_integer_value(pair.first, pair.second.data()));
+        }
+    }
+    client_config->add_reliable_topic_config(config);
+}
+void
+client_dom_config_processor::handle_near_cache(
+  const boost::property_tree::ptree& node)
+{
+    handle_near_cache_node(node);
+}
+void
+client_dom_config_processor::handle_near_cache_node(
+  const boost::property_tree::ptree& node) const
+{
+    std::string name;
+    try {
+        name = get_attribute(node, "name");
+    } catch (const boost::exception& e) {//if name is not explicitly specified then name = default
+        name = "default";
+    }
+    hazelcast::client::config::near_cache_config near_cache_config(name);
+    for (auto& pair : node) {
+        auto child = pair_to_node(pair.first, pair.second);
+        std::string node_name = pair.first;
+        if (matches("time-to-live-seconds", node_name)) {
+            near_cache_config.set_time_to_live_seconds(
+              get_integer_value(pair.first, pair.second.data()));
+        } else if (matches("max-idle-seconds", node_name)) {
+            near_cache_config.set_max_idle_seconds(
+              get_integer_value(pair.first, pair.second.data()));
+        } else if (matches("in-memory-format", node_name)) { // NO NATIVE ??
+            if (pair.second.data() == "BINARY") {
+                near_cache_config.set_in_memory_format(
+                  hazelcast::client::config::in_memory_format::BINARY);
+            } else if (pair.second.data() == "OBJECT") {
+                near_cache_config.set_in_memory_format(
+                  hazelcast::client::config::in_memory_format::OBJECT);
+            }
+        } else if (matches("invalidate-on-change", node_name)) {
+            near_cache_config.set_invalidate_on_change(
+              get_bool_value(pair.second.data()));
+        } else if (matches("local-update-policy", node_name)) {
+            if (pair.second.data() == "CACHE") {
```

Comment:
'CACHE' is not a correct value, it should be changed to 'CACHE_ON_UPDATE' according to java implementation.